core/node/libp2p/addrs.go

```diff
@@ -25,14 +25,23 @@ func AddrFilters(filters []string) func() (*ma.Filters, Libp2pOpts, error) {
 	}
 }
 
-func makeAddrsFactory(announce []string, noAnnounce []string) (p2pbhost.AddrsFactory, error) {
-	var annAddrs []ma.Multiaddr
-	for _, addr := range announce {
-		maddr, err := ma.NewMultiaddr(addr)
+func makeAddrsFactory(announce []string, appendAnnouce []string, noAnnounce []string) (p2pbhost.AddrsFactory, error) {
+	var err error // To assign to the slice in the for loop
+
+	annAddrs := make([]ma.Multiaddr, len(announce))
+	for i, addr := range announce {
+		annAddrs[i], err = ma.NewMultiaddr(addr)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	appendAnnAddrs := make([]ma.Multiaddr, len(appendAnnouce))
+	for i, addr := range appendAnnouce {
+		appendAnnAddrs[i], err = ma.NewMultiaddr(addr)
```

Comment:
What's the intended behavior if both announce and appendannounce are set? Do we just need to quietly ensure that we don't duplicate addresses, but that otherwise it's fine to set both fields?

Reply:
I believe deduplicating should be enough. 

My mental model is:
- `Announce` is "ignore infered ones, use these" 
-  `AppendAnnounce` is "add these to the final list, no matter if static or inferred"

Reply:
I think the same thing as @lidel

Reply:
Added deduplication in https://github.com/ipfs/go-ipfs/pull/8177/commits/ed1edb6e8d7bfe9891302a70ab6c6bd9adc6d11e